payjoin/src/uri.rs

```diff
@@ -27,8 +28,8 @@ pub struct PayjoinParams {
     pub(crate) disable_output_substitution: bool,
 }
 
-pub type Uri<'a> = bip21::Uri<'a, Payjoin>;
-pub type PjUri<'a> = bip21::Uri<'a, PayjoinParams>;
+pub type Uri<'a> = bip21::Uri<'a, NetworkUnchecked, Payjoin>;
+pub type PjUri<'a> = bip21::Uri<'a, NetworkChecked, PayjoinParams>;
```

Comment:
The URIs need to be generic over network validation.

Reply:
please elaborate why? This works in both test and the downstream binaries

Reply:
Because server creates Uri which is already checked. But it just occurred to me that server always creates `PjUri` rather than `Uri` so `PjUri` could be always `NetworkChecked` while `Uri` could be generic. So people would first call `.require_network()?` and then `check_pj_supported()`